static/src/javascripts/lib/detect.js

```diff
@@ -270,7 +270,7 @@ const adblockInUse = getAdblockInUse();
 
 const getReferrer = () => document.referrer || '';
 
-const getUserAgent = (() => {
+const getUserAgent = () => {
```

Comment:
We change the type of this here from `string` to `() => string`. This makes it much easier to test, and I think more flexible about where you use it.

I assume the original reason for calling the function immediately was to only compute the user agent once and not every time you wanted it, so I've included a `userAgent` that is computed immediately and exported. This is used in two JS files than I'm keen not to mess with too much.